shuttle_xpress/src/shuttle_xpress/shuttle_xpress

```diff
@@ -0,0 +1,144 @@
+        """
+        Grabs the usb device.
+        """
+        self.__device = usb.core.find(idVendor=CONTOUR_ID_VENDOR,
+                                      idProduct=CONTOUR_ID_PRODUCT)
+
+        if self.__device is None:
+            raise ValueError("Contour Shuttle Xpress was not found")
+
+        if self.__device.is_kernel_driver_active(self.__interface):
+            rospy.logdebug("Kernel driver attached - trying to detach the driver.")
+            try:
+                self.__device.detach_kernel_driver(self.__interface)
+                rospy.logdebug("kernel driver detached")
+            except usb.core.USBError as usb_exc:
+                sysexit("Could not detach kernel driver: %s" %str(usb_exc))
+
+        self.__device.set_configuration()
+        usb.util.claim_interface(self.__device, self.__interface)
+
+    def release_device(self):
+        """
+        Releases the devices and reattaches the kernel driver if necessary.
+        """
+        try:
+            usb.util.release_interface(self.__device, self.__interface)
+            self.__device.attach_kernel_driver(self.__interface)
+        except usb.core.USBError:
+            rospy.logwarn("Didn't reattach the kernel driver properly.")
+
+    def read_data(self):
+        """
+        Reads the data - called from a thread.
+        """
+        #Reading the data
+        endpoint = self.__device[0][(0, 0)][0]
+        while not rospy.is_shutdown():
+            try:
+                data = self.__device.read(endpoint.bEndpointAddress,
+                                          endpoint.wMaxPacketSize)
+
+                self._fill_state(data)
+            except usb.core.USBError:
+                rospy.logdebug("Reader timedout - no events received.")
+
+    def _fill_state(self, data):
+        """
+        Fills the self.__state properly from the raw data.
+        """
+        wheel = data[0]
+        if wheel > 127:
+            wheel -= 254
+
+        self.__state.shuttle_wheel = wheel
+        self.__state.rotating_dial = data[1]
+
+        # the first for buttons are using binary bit masks
+        for i in range(4):
+            if self.__check_btn(data[3], i):
+                self.__state.buttons[i] = True
+            else:
+                self.__state.buttons[i] = False
+
+        # the last button is in the last data - not using a bit check
+        if data[4]:
+            self.__state.buttons[4] = True
+        else:
+            self.__state.buttons[4] = False
+
+        self._pub.publish(self.__state)
+
+    def __check_btn(self, raw_data, btn_index):
+        """
+        returns a positive value if the bit at btn_index in raw_data is set.
+        """
+        raw_data = raw_data >> 4 # for some reason the buttons start at bit 4
+        btn_index = 1 << btn_index
+
+        return raw_data & btn_index
+
+if __name__ == "__main__":
+    rospy.init_node("shuttle_xpress")
+
+    with ShuttleXpressUsb() as shuttle:
+        rospy.spin()
```

Comment:
I think this should be `shuttle.read_data()`?

Otherwise we have a loop inside the constructor, which is weird. And then when we exit the constructor we spin().